### ISSUE TITLE
Merge all ACTION operations into a single list

### ISSUE THREAD

**Issue by Kazuo256** [uspgamedev/backdoor#307]

**Comment by Kazuo256** [comment]
Alright, I started taking a look at this and found problem: canceling (or failing) actions midway through the list of operations. Rollback is too hard, and lazy evaluation doesn't solve all cases.

Thinking back, this might've been one of the reasons why I separated actions into INPUTS, OPERATIONS, and EFFECTS in the first place. Because, essentially, all INPUTS *must* happen before the first EFFECT takes place.

An alternate solution is to carry on with the unification of INPUTS, OPERATIONS, and EFFECTS into only OPERATIONS, but also add a ternary tag to indicate whether they require input or change the state of the game (or none of the above). Then we can invalidate actions where inputs happen after the first effect.

We can still do nothing for failing actions though. Then again, a simple crash sounds healthy enough.

**Comment by Kazuo256** [comment]
Solution: merge only operators and effects.

**Comment by rilifon** [comment]
This solves the problem of canceling actions? How about AI behaviour?

**Comment by Kazuo256** [comment]
> This solves the problem of canceling actions?

Yes, because parameters are the only "cancellable" part, and they would be handled separately (before operators and effects alike, just like they are now).

> How about AI behaviour?

Should stay the same: they choose an action and provide the corresponding parameters.
